src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Simplification/Simplifiers/CastSimplifier.cs

```diff
@@ -461,7 +461,11 @@ private static bool CastRemovalCouldCauseSignExtensionWarning(ExpressionSyntax c
                 return true;
 
             // We can safely remove convertion to object in interpolated strings regardless of nullability
-            if (castNode.IsParentKind(SyntaxKind.Interpolation) && originalConversionOperation.Type?.SpecialType is SpecialType.System_Object)
+            if (castNode.IsParentKind(SyntaxKind.Interpolation) && originalConvertedType.SpecialType is SpecialType.System_Object)
+                return true;
+
+            // Can remove convertion of string literal to string regardless of nullability
+            if (rewrittenExpression.WalkDownParentheses().IsKind(SyntaxKind.StringLiteralExpression) && originalConvertedType.SpecialType is SpecialType.System_String)
```

Comment:
Why is `string` special here? Unless I'm missing something, I think this needs a generalized approach, for example, `var c = new C(); M(c);`

Reply:
Now when you've said it I am wondering if it is ok to simplify any cast of non-nullable to nullable when the type is a reference type🤔